lib/plugins/ray_trace.js

```diff
@@ -1,28 +1,35 @@
-const Vec3 = require('vec3')
+const { Vec3 } = require('vec3')
+const { RaycastIterator } = require('../iterators')
 
-module.exports = inject
-
-function inject (bot) {
-  const rayTraceBlock = (maxSteps = 256, vectorLength = 5 / 16) => {
-    const { height, position, yaw, pitch } = bot.entity
-    const cursor = position.offset(0, height, 0)
+module.exports = (bot) => {
+  function getViewDirection (pitch, yaw) {
+    const csPitch = Math.cos(pitch)
+    const snPitch = Math.sin(pitch)
+    const csYaw = Math.cos(yaw)
+    const snYaw = Math.sin(yaw)
+    return new Vec3(-snYaw * csPitch, snPitch, -csYaw * csPitch)
+  }
 
-    const x = -Math.sin(yaw) * Math.cos(pitch)
-    const y = Math.sin(pitch)
-    const z = -Math.cos(yaw) * Math.cos(pitch)
+  bot.blockInSight = (maxSteps = 256, vectorLength = 5 / 16) => {
+    console.warn('[deprecated] use bot.blockAtCursor instead')
```

Comment:
Hmmm... I'm not a fan of `console.*` methods. It would break `mineflayer-dashboard` if user does not reassign it to the dashboard's log function ;/

Also, in my opinion we should call it only once.

Reply:
What do you suggest then? The goal is to really drop the old api as it makes no sense to keep it, other that to ensure a backward compatibility period. If it breaks some code, at least it will put a warning to tell the user what to do to fix it.

Reply:
I understand that. Leave it, I'll just push a patch that overwrites this function and logs into the dashboard log.